### PR TITLE
Add mutable/immutable java.util.List wrappers

### PR DESCRIPTION
**Pull request by seanf**
Based on the work in PR https://github.com/javaslang/javaslang/pull/1736 for (part of) issue https://github.com/javaslang/javaslang/issues/1728.

I suggest reviewing this PR via reviewable (it will add comments here in github): https://reviewable.io/reviews/javaslang/javaslang/1751. I generally find it makes it easier to keep track of individual concerns in the code.

### CODE REVIEW

File: javaslang/src/main/java/javaslang/collection/JavaConverters.java [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+        }
+
+        public Seq<Object> narrowed() {
+            return Seq.narrow(delegate);
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
```
Comment (sputnikci) [inline javaslang/src/main/java/javaslang/collection/JavaConverters.java]: \[PMD\] INFO: Ensure you override both equals\(\) and hashCode\(\)

Comment (seanf) [inline javaslang/src/main/java/javaslang/collection/JavaConverters.java]: Can we turn this off in PMD? Or should I just call super.hashCode()? Because that's the implementation I want to use. I considered inheriting super.equals too, but this implementation is probably more efficient.

**Comment by danieldietrich** [comment]
@seanf @paplorinc I'm thankful for all your effort but this is too much for me :-) We now have two PRs open for the same issue and they diverge somehow.

The naming changed here, I created code that was not parameterized with an interface and so on. It is too much effort for me to review on that detail level on two diverging PRs.

Beside that I've created already an own implementation in order to be able to give review code examples. This is a third impl for that issue now.

Long story short: I want to finish my impl with your suggestions and close the existing two PRs in order to reduce the review noise and time effort.

That's the first time I do that and I hope not to piss off anyone...

---

Side-note: For now we focus on using plain Github for reviews.

**Comment by seanf** [comment]
Closed in favour of https://github.com/javaslang/javaslang/pull/1752

**Comment by paplorinc** [comment]
Thanks @seanf :)
